src/main/java/org/webshar/hrms/model/db/ResourcePermission.java

```diff
@@ -0,0 +1,76 @@
+package org.webshar.hrms.model.db;
+
+import java.io.Serializable;
+import java.util.Date;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Table;
+import javax.persistence.UniqueConstraint;
+import lombok.Getter;
+import lombok.Setter;
+import org.hibernate.annotations.CreationTimestamp;
+import org.hibernate.annotations.UpdateTimestamp;
+import org.springframework.stereotype.Component;
+
+@Component
+@Entity
+@Table(name = "resource_permission", uniqueConstraints = @UniqueConstraint(columnNames = {"id"}))
+public class ResourcePermission implements Serializable
```

Comment:
Considering one resource can have many permissions, I think instead of creating a new entity here, modify the Resource entity to include List<Permission> to model one-to-many relationship. 
e.g.
class Resource{
@OnetoMany
List<Permission> permissions
} 
Explore this option.
JPA will create a join table in this case with resource id and permission id columns.

Reply:
Yes we also have that thought but again this is just at CRUD implementation for storing/updating relation of resource and permission for that we added that entity.

Reply:
If the resource has permissions attribute, then crud operations on resources will cover permissions as well. And the join table which got generated due to one to many mapping and respective crud operations are managed by JPA itself. You do not have to create an entity, repo, service for that part.